File: src/main/kotlin/net/zhuruoling/omms/controller/mirai/network/Target.java
package net.zhuruoling.omms.controller.mirai.network;

public class Target {
    String address;
    int port;

    public Target(String ip, int port) {
        this.address = ip;
        this.port = port;
    }

    public int getPort() {
        return port;
    }

    public void setPort(int port) {
        this.port = port;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Override
    public int hashCode() {
        return (address + port).hashCode();
    }

    @Override
    public String toString() {
        return "Target{" +
                "address='" + address + '\'' +
                ", port=" + port +
                '}';
    }
}

File: src/main/kotlin/net/zhuruoling/omms/controller/mirai/network/Target.kt
package net.zhuruoling.omms.controller.mirai.network

class Target(var address: String, var port: Int) {

    override fun hashCode(): Int {
        return (address + port).hashCode()
    }

    override fun toString(): String {
        return "Target{" +
            "address='" + address + '\'' +
            ", port=" + port +
            '}'
    }

    fun port(): Int {
        return port
    }

    fun address(): String {
        return address
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Target

        if (address != other.address) return false
        if (port != other.port) return false

        return true
    }
}